TASK-58584: Left chatrooms are still displayed in favorite list

akhanfir opened the pull request:
Prior to this change, when I leave the chat room the status of his favorite remains himself.
To fix this, disabled the favorite when leaving the chat room.

akhanfir committed:
TASK-58584: Left chatrooms are still displayed in favorite list
Prior to this change, when I leave the chat room the status of his favorite remains himself.
To fix this, disabled the favorite when leaving the chat room.

ahamdi commented on application/src/main/webapp/vue-app/components/ExoChatRoomDetail.vue:
```diff
@@ -262,6 +262,7 @@ export default {
         this.confirmKOMessage = settingAction.confirm.koMessage;
         this.confirmAction = settingAction.confirm.confirmed;
         this.showConfirmModal = true;
+        document.dispatchEvent(new CustomEvent(chatConstants.ACTION_ROOM_FAVORITE_REMOVE, {'detail': this.contact}));
```
This should be fired just in case of removing favorite action, right ?

akhanfir committed:
Left chatrooms are still displayed in favorite list
Prior to this change, when leave the room marcked is favorite the left chatroom list from big chat app marked as favorite is still displayed

To fix this, when click to leave the room which was already marked as favorite I call it the event to delete this room in the list of favorites

ahamdi commented on application/src/main/webapp/vue-app/components/ExoChatApp.vue:
```diff
@@ -38,7 +38,7 @@
     </div>
     <div
-      v-show="(selectedContact && (selectedContact.room || selectedContact.user)) || mq === 'mobile'"
+      v-show="(isSelectedContact && selectedContact && (selectedContact.room || selectedContact.user)) || mq === 'mobile'"
```
this is unneeded IMO ?
```suggestion
      v-show="(selectedContact && (selectedContact.room || selectedContact.user)) || mq === 'mobile'"
```
